lib/spack/spack/database.py

```diff
@@ -95,30 +95,33 @@ class InstallRecord(object):
 
     """
 
-    def __init__(self, spec, path, installed, ref_count=0, explicit=False):
+    def __init__(self, spec, path, installed, ref_count=0, explicit=False, external=False):  # NOQA: ignore=E501
         self.spec = spec
         self.path = str(path)
         self.installed = bool(installed)
         self.ref_count = ref_count
         self.explicit = explicit
+        self.external = external
 
     def to_dict(self):
         return {
             'spec': self.spec.to_node_dict(),
             'path': self.path,
             'installed': self.installed,
             'ref_count': self.ref_count,
-            'explicit': self.explicit
+            'explicit': self.explicit,
+            'external': self.external
         }
 
     @classmethod
     def from_dict(cls, spec, dictionary):
         d = dictionary
         return InstallRecord(spec, d['path'], d['installed'], d['ref_count'],
-                             d.get('explicit', False))
+                             d['explicit'], d['external'])
```

Comment:
@mathstuf Apparently not... `spec` may be in `dictionary`, so I had to resort to popping it out before doing a `**d`. Would this be good enough ?